src/ambulance_game/markov/waiting.py

```diff
@@ -303,56 +297,63 @@ def mean_waiting_time_formula(
     threshold : int
     system_capacity : int
     buffer_capacity : int
-    formula : str, optional
+    waiting_formula : function
 
     Returns
     -------
     float
-        The mean waiting time for the specified class
+        The overall mean waiting time by combining class 1 and class 2 individuals
     """
-    if formula == "recursive":
-        mean_waiting_time = mean_waiting_time_formula_using_recursive_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
-
-    if formula == "algebraic":
-        mean_waiting_time = mean_waiting_time_formula_using_algebraic_approach(
-            all_states=all_states,
-            pi=pi,
-            class_type=class_type,
-            lambda_2=lambda_2,
-            lambda_1=lambda_1,
-            mu=mu,
-            num_of_servers=num_of_servers,
-            threshold=threshold,
-            system_capacity=system_capacity,
-            buffer_capacity=buffer_capacity,
-        )
+    mean_waiting_time_class_1 = waiting_formula(
+        all_states=all_states,
+        pi=pi,
+        class_type=1,
+        lambda_2=lambda_2,
+        lambda_1=lambda_1,
+        mu=mu,
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        buffer_capacity=buffer_capacity,
+    )
+    mean_waiting_time_class_2 = waiting_formula(
+        all_states=all_states,
+        pi=pi,
+        class_type=2,
+        lambda_2=lambda_2,
+        lambda_1=lambda_1,
+        mu=mu,
+        num_of_servers=num_of_servers,
+        threshold=threshold,
+        system_capacity=system_capacity,
+        buffer_capacity=buffer_capacity,
+    )
```

Comment:
Not sure why GitHub won't let me suggest (possibly because this is a commit comment?):

```
    mean_waiting_times = [waiting_formula(
         all_states=all_states,
         pi=pi,
         class_type=class_type,
         lambda_2=lambda_2,
         lambda_1=lambda_1,
         mu=mu,
         num_of_servers=num_of_servers,
         threshold=threshold,
         system_capacity=system_capacity,
         buffer_capacity=buffer_capacity,
      )

      for class_type in (1, 2)]
```

Note that it's probably worth changing the class types in code to be 0 and 1 (in text/mathematics keep them as 1 and 2): there will be a number of places later on where the indexing will take care of things. Like for example:

```
mean_waiting_times[0]
```

```
mean_waiting_times[1]
```

Reply:
That looks good. But I'm not entirely sure I like the idea of changing class types to 0 and 1 (in terms of being consistent with the maths as well). I will definitely apply the suggestion but maybe leave it as (1,2). Is that okay?

Reply:
It's written in Python and Python starts indexing at 0 so it'll make things a lot simpler throughout.